prefect_gcp/cloud_run_job.py

```diff
@@ -0,0 +1,686 @@
+from prefect.utilities.asyncutils import run_sync_in_worker_thread, sync_compatible
+from pydantic import BaseModel, Field, root_validator, validator
+
+from prefect_gcp.credentials import GcpCredentials
+
+
+class Job(BaseModel):
+    """
+    Utility class to call GCP `jobs` API and
+    interact with the returned objects.
+    """
+
+    metadata: dict
+    spec: dict
+    status: dict
+    name: str
+    ready_condition: dict
+    execution_status: dict
+
+    def _is_missing_container(self):
+        """
+        Check if Job status is not ready because
+        the specified container cannot be found.
+        """
+        if (
+            self.ready_condition.get("status") == "False"
+            and self.ready_condition.get("reason") == "ContainerMissing"
+        ):
+            return True
+        return False
+
+    def is_ready(self) -> bool:
+        """Whether a job is finished registering and ready to be executed"""
+        if self._is_missing_container():
+            raise Exception(f"{self.ready_condition['message']}")
+        return self.ready_condition.get("status") == "True"
+
+    def has_execution_in_progress(self) -> bool:
+        """See if job has a run in progress."""
+        return (
+            self.execution_status == {}
+            or self.execution_status.get("completionTimestamp") is None
+        )
+
+    @staticmethod
+    def _get_ready_condition(job: dict) -> dict:
+        """Utility to access JSON field containing ready condition."""
+        if job["status"].get("conditions"):
+            for condition in job["status"]["conditions"]:
+                if condition["type"] == "Ready":
+                    return condition
+
+        return {}
+
+    @staticmethod
+    def _get_execution_status(job: dict):
+        """Utility to access JSON field containing execution status."""
+        if job["status"].get("latestCreatedExecution"):
+            return job["status"]["latestCreatedExecution"]
+
+        return {}
+
+    @classmethod
+    def get(cls, client: Resource, namespace: str, job_name: str):
+        """Make a get request to the GCP jobs API and return a Job instance."""
+        request = client.jobs().get(name=f"namespaces/{namespace}/jobs/{job_name}")
+        response = request.execute()
+
+        return cls(
+            metadata=response["metadata"],
+            spec=response["spec"],
+            status=response["status"],
+            name=response["metadata"]["name"],
+            ready_condition=cls._get_ready_condition(response),
+            execution_status=cls._get_execution_status(response),
+        )
+
+    @classmethod
+    def create(cls, client: Resource, namespace: str, body: dict):
+        """Make a create request to the GCP jobs API."""
+        request = client.jobs().create(parent=f"namespaces/{namespace}", body=body)
+        response = request.execute()
+        return response
+
+    @classmethod
```

Comment:
This doesn't use `cls` and could be a static method
```suggestion
    @staticmethod
```